File: packetCapture/src/main/java/com/ninpou/packetcapture/core/util/common/TimeFormatter.java
package com.ninpou.packetcapture.core.util.common;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class TimeFormatter {
    private static DateFormat HHMMSSSFormat = new SimpleDateFormat("HH:mm:ss:s",
            Locale.getDefault());
    private static DateFormat formatYYMMDDHHMMSSFormat = new SimpleDateFormat("yyyy:MM:dd-HH:mm:ss:s",
            Locale.getDefault());

    public static String formatToHHMMSSMM(long time) {
        Date date = new Date(time);
        return HHMMSSSFormat.format(date);
    }

    public static String formatToYYMMDDHHMMSS(long time) {
        Date date = new Date(time);
        return formatYYMMDDHHMMSSFormat.format(date);
    }
}

File: packetCapture/src/main/java/com/ninpou/packetcapture/core/util/common/TimeFormatter.kt
package com.ninpou.packetcapture.core.util.common

import java.text.DateFormat
import java.text.SimpleDateFormat
import java.util.*

//Locale fix for inconsistent state
object TimeFormatter {
    private val HHMMSSSFormat: DateFormat = SimpleDateFormat("HH:mm:ss:s", Locale.getDefault())
    private val formatYYMMDDHHMMSSFormat: DateFormat = SimpleDateFormat("yyyy:MM:dd-HH:mm:ss:s", Locale.getDefault())

    @JvmStatic
    fun formatToHHMMSSMM(time: Long): String {
        val date = Date(time)
        return HHMMSSSFormat.format(date)
    }

    @JvmStatic
    fun formatToYYMMDDHHMMSS(time: Long): String {
        val date = Date(time)
        return formatYYMMDDHHMMSSFormat.format(date)
    }
}